File: code/csharp-visualstudio/Practice/02-FarmModel.cs
using System;
using Google.OrTools.LinearSolver;

namespace Practice
{
  public class FarmModel
  {
    public static void Solve()
    {
      // Create an instance of the Solver class
      var solver = Solver.CreateSolver("FarmModel", "GLOP_LINEAR_PROGRAMMING");

      // Create variables for each type of Anaimal

      // Create expression for Revenue
      
      // Add the expression to the solver
      

      // Create constraint for Pasture
      

      // Create constraint for the Feed
      

      // Create constraint for the Labor
      

      // Solve the problem
      

      //// Evaluate the solve status
      //if (resultStatus != Solver.ResultStatus.OPTIMAL)
      //{
      //  Console.WriteLine("The problem does not have an optimal solution!");
      //  return;
      //}

      //// Print the results
      //Console.WriteLine("Solution:");
      //Console.WriteLine("Objective value = " + solver.Objective().Value());
      //foreach (var variable in solver.variables())
      //{
      //  Console.WriteLine($"{variable.Name()} = {variable.SolutionValue()}");
      //}
    }
  }
}

File: code/fsharp-visualstudio/Practice/02-FarmModel.fs
module Practice.FarmModel

open Flips
open Flips.Types

let solve () =

    // Create decision variables for cows, pigs, and chickens

    // Create an objective expression

    // Create an objective


    // Add constraint for amount of Pasture

    // Add constraint for amount of Feed

    // Add constraint for amount of Labor


    // Create the model and add the constraints


    // Create the solver settings
    let settings = {
        SolverType = SolverType.CBC
        MaxDuration = 10000L
        WriteLPFile = None
    }

    // Solve the model and save the results

    printfn "--Results--"

    // Print the results of the solver
    //match result with
    //| Optimal solution -> 
    //    printfn "Objective Value: %f" solution.ObjectiveResult
    //    for (decision, value) in solution.DecisionResults |> Map.toSeq do
    //        let (DecisionName name) = decision.Name
    //        printfn "Decision: %s\tValue: %f" name value
    //| _ -> printfn "Unable to solve."